k2/csrc/fsa_utils.cu

```diff
@@ -814,4 +814,103 @@ Ragged<int32_t> GetIncomingArcs(FsaVec &fsas,
       incoming_arcs_order);
 }
 
+Ragged<int32_t> GetLeavingArcIndexBatches(FsaVec &fsas,
+                                          Ragged<int32_t> &state_batches) {
+  K2_CHECK_EQ(fsas.NumAxes(), 3);
```

Comment:
Do we need to check the number of axes with `state_batches`?

Reply:
Added